models/trie_mimc/frontend_proof.go

```diff
@@ -0,0 +1,477 @@
+package trie_mimc
+
+import (
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/std/hash/mimc"
+)
+
+// Return the result of right shift by 1, input size = 32
```

Comment:
Should be `Return the result of right shift by 1 byte, input size = 32 bytes`
32 byte is fixed (in the compiled circuit the size is fixed). It can be less (by leading 0s) but it will be regarded as 32 bytes always

Reply:
Thanks for pointing this out. Addressed this by [0137176].(https://github.com/iotaledger/trie.go/pull/1/commits/0137176a578e4cf74047b93b897b22cca9fa5236)